### DISCUSSION TITLE
Not able to run all the services

### DISCUSSION THREAD

**Discussion by vigneshs-dev** [meysamhadeli/shop-golang-microservices#99] [General]
Hi @meysamhadeli ,
I followed the steps which you have given to build the services separately. But I am facing errors. Can you please provide steps clearly to run all the services separately and connect them using rabbitmq ?

**Comment by meysamhadeli** [comment]
> Hi @meysamhadeli , I followed the steps which you have given to build the services separately. But I am facing errors. Can you please provide steps clearly to run all the services separately and connect them using rabbitmq ?

Hi @vigneshs-dev, 
For running the project you need to follow the [How to Run
](https://github.com/meysamhadeli/shop-golang-microservices?tab=readme-ov-file#how-to-run) steps.
1- First step, we need run `docker compose` `infrastructure.yaml` for installing all of infrastructure that we need.
2- Second step, in `root of project` with this command `go build -v ./...`, we can build whole of services.
3- And other steps like documentation for `running` project...

If you follow steps like doc and have error in specific step, please give me more details or screen shot from problem here.
Thanks :)

**Comment by vigneshs-dev** [comment]
I followed the steps given by you. First I ran the docker-compose file. All the three containers are created and running in my system. Then I build and run the identity services. But I got the error which related to postgres password. I will attach the screenshot for your kind reference.

![image](https://github.com/user-attachments/assets/3c996b82-c7f2-428d-abdb-813300623abd)

**Reply by meysamhadeli** [reply]
Hi @vigneshs-dev, I had a typo in docker compose `infrastructure.yaml` and in postgres section POSTGRES_USER and POSTGRES_PASSWORD had typo. I fixed the problem and now you can checked the fixed deployment here: [infrastructure.yaml](https://github.com/meysamhadeli/shop-golang-microservices/blob/main/deployments/docker-compose/infrastructure.yaml).
I fixed values from postgess to postgres and now everything should works properly.
